src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -173,15 +179,159 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                         if (InitializationCacheExpires < DateTimeOffset.UtcNow)
                         {
                             InitializationCacheExpires = DateTimeOffset.UtcNow.Add(MinCacheExpirationInterval);
-                            await LoadAll(ignoreFailures: false, cancellationToken).ConfigureAwait(false);
+                            await LoadAllWithFailOverPolicyAsync(ignoreFailures: false, cancellationToken).ConfigureAwait(false);
                         }
 
                         return;
                     }
 
-                    await RefreshIndividualKeyValues(cancellationToken).ConfigureAwait(false);
-                    await RefreshKeyValueCollections(cancellationToken).ConfigureAwait(false);
-                    await RefreshKeyValueAdapters(cancellationToken).ConfigureAwait(false);
+                    //
+                    // Avoid instance state modification
+                    Dictionary<string, ConfigurationSetting> applicationSettings = null;
+                    Dictionary<KeyValueWatcher, KeyValueChange> keyValueChanges = null;
+                    List<KeyValueChange> changedKeyValuesCollection = null;
+                    bool refreshAll = false;
+
+                    await ExecuteWithFailOverPolicyAsync(async (client) =>
+                        {
+                            keyValueChanges = new Dictionary<KeyValueWatcher, KeyValueChange>();
+                            refreshAll = false;
+                            changedKeyValuesCollection = null;
+                            applicationSettings = null;

```

Comment:
NIT: for easier reading, do you mind initializing these variables in the same order as they are declared outside?

Reply:
Done!